ReactAndroid/src/main/java/com/facebook/react/modules/fresco/ReactOkHttpNetworkFetcher.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ * <p/>
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.modules.fresco;
+
+import android.net.Uri;
+import android.os.SystemClock;
+
+import com.facebook.imagepipeline.backends.okhttp3.OkHttpNetworkFetcher;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.ReadableMapKeySetIterator;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+import okhttp3.CacheControl;
+import okhttp3.Headers;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+
+class ReactOkHttpNetworkFetcher extends OkHttpNetworkFetcher {
+
+  private static final String TAG = "ReactOkHttpNetworkFetcher";
+
+  private final OkHttpClient mOkHttpClient;
+  private final Executor mCancellationExecutor;
+
+  /**
+   * @param okHttpClient client to use
+   */
+  public ReactOkHttpNetworkFetcher(OkHttpClient okHttpClient) {
+    super(okHttpClient);
+    mOkHttpClient = okHttpClient;
+    mCancellationExecutor = okHttpClient.dispatcher().executorService();
+  }
+
+  private Map<String, String> getHeaders(ReadableMap readableMap) {
+    if (readableMap == null) {
+        return null;
+    }
+    ReadableMapKeySetIterator iterator = readableMap.keySetIterator();
+    Map<String, String> map = new HashMap<>();
+    while (iterator.hasNextKey()) {
+      String key = iterator.nextKey();
+      String value = readableMap.getString(key);
+      map.put(key, value);
+    }
+    return map;
+  }
+
+  @Override
+  public void fetch(final OkHttpNetworkFetchState fetchState, final Callback callback) {
+    fetchState.submitTime = SystemClock.elapsedRealtime();
+    final Uri uri = fetchState.getUri();
+    Map<String, String> requestHeaders = null;
+    if (fetchState.getContext().getImageRequest() instanceof ReactNetworkImageRequest) {
+      ReactNetworkImageRequest networkImageRequest = (ReactNetworkImageRequest)
+        fetchState.getContext().getImageRequest();
+      requestHeaders = getHeaders(networkImageRequest.getHeaders());
+    }
+    if (requestHeaders == null) {
+      requestHeaders = Collections.emptyMap();
+    }
+    final Request request = new Request.Builder()
+      .cacheControl(new CacheControl.Builder().noStore().build())
```

Comment:
@foghina My understanding is that React Native is currently applying `no-store` to all requests because that is what Fresco is doing. Here's how I came to that conclusion:
- `FrescoModule` uses `OkHttpImagePipelineConfigFactory` ([code](https://github.com/facebook/react-native/blob/be5e300abf9aee0f471c08f80d515c2b6f221eeb/ReactAndroid/src/main/java/com/facebook/react/modules/fresco/FrescoModule.java#L117))
- `OkHttpImagePipelineConfigFactory` calls `setNetworkFetcher` with `OkHttpNetworkFetcher` ([code](https://github.com/facebook/fresco/blob/5f9722e56d6d4c5b7382eca6af8f11449a4fd43a/imagepipeline-backends/imagepipeline-okhttp3/src/main/java/com/facebook/imagepipeline/backends/okhttp3/OkHttpImagePipelineConfigFactory.java#L24))
- `OkHttpNetworkFetcher` applies `no-store` to all requests ([code](https://github.com/facebook/fresco/blob/e437ef96c5c63e10c4dd6a94bd44cf58e4d7118d/imagepipeline-backends/imagepipeline-okhttp3/src/main/java/com/facebook/imagepipeline/backends/okhttp3/OkHttpNetworkFetcher.java#L90))

You raise an interesting question but I set `no-store` on all requests because I was trying to preserve the current behavior. I don't have background on why things are done that way. If I am right that this is the current behavior, then I think the issue you've raised is tangential to my PR.